src/sidebar/services/view-filter.js

```diff
@@ -100,9 +100,8 @@ function viewFilter(unicode) {
     quote: {
       autofalse: ann => (ann.references || []).length > 0,
       value(annotation) {
-        if (!annotation.target) {
-          // FIXME: All annotations *must* have a target, so this check should
-          // not be required.
+        if (!annotation.id) {
```

Comment:
Reading this code it wasn't immediately obvious why we're testing for `annotation.id` but the following code doesn't access `annotation.id` at all. Looking at the value of a new not-yet-saved annotation, it seems the issue is that the `target` array is empty, whereas this code previously assumed it was either not present or non-empty if present.

<img width="784" alt="Screenshot 2019-08-13 15 05 18" src="https://user-images.githubusercontent.com/2458/62948254-f062b900-bddb-11e9-98bf-45d19c95284c.png">

Related to this, I notice this code partly duplicates some logic from the [annotation component](https://github.com/hypothesis/client/blob/2334426497bcf4c930ba9deae91a5d03ab35616c/src/sidebar/components/annotation.js#L331). In future we could extract that into a helper.